src/ch00-06-maps.md

```diff
@@ -0,0 +1,18 @@
+# Maps
+
+Maps are a key-value data structure used to store data within a smart contract. In Cairo they are implemented using the `LegacyMap` type. 
```

Comment:
can you specify here that the `LegacyMap` type can only be used inside the `Storage` struct of a contract, and that it can't be used elsewhere? I think it's important enough to be moved from the `notes` section to the first paragraph.